### PR TITLE
deploy: deploy erasure coded pool

### PR DESCRIPTION
**Pull request by Yuggupta27**
add functionality to deploy erasure-coded 
pool during cluster deployment to allow usage and testing
in erasure-coded pools.

Signed-off-by: Yug Gupta <yuggupta27@gmail.com>

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

File: scripts/rook.sh [hunk 1]
```diff
@@ -4,6 +4,7 @@ ROOK_VERSION=${ROOK_VERSION:-"v1.6.2"}
 ROOK_DEPLOY_TIMEOUT=${ROOK_DEPLOY_TIMEOUT:-300}
 ROOK_URL="https://raw.githubusercontent.com/rook/rook/${ROOK_VERSION}/cluster/examples/kubernetes/ceph"
 ROOK_BLOCK_POOL_NAME=${ROOK_BLOCK_POOL_NAME:-"newrbdpool"}
+ROOK_EC_POOL_NAME=${ROOK_EC_POOL_NAME:-"ec-pool"}
```
Comment (Madhu-1) [inline scripts/rook.sh]: add `block`  to all the variables and options as its a specific to block

Comment (Madhu-1) [inline scripts/rook.sh]: `ROOK_EC_POOL_NAME` to `ROOK_EC_BLOCK_POOL_NAME`

File: scripts/rook.sh [hunk 2]
```diff
@@ -116,6 +117,22 @@ function delete_block_pool() {
 	rm -f "./newpool.yaml"
 }
 
+function create-block-ec-pool() {
+	curl -o pool-ec.yaml "${ROOK_URL}/pool-ec.yaml"
```
Comment (Madhu-1) [inline scripts/rook.sh]: update filename to refect the block name

Comment (Madhu-1) [reply scripts/rook.sh]: files name are different in rook. this should be fine

**Review by Madhu-1 — DISMISSED** [review]

**Review by Madhu-1 — DISMISSED** [review]

File: scripts/rook.sh [hunk 3]
```diff
@@ -116,6 +117,22 @@ function delete_block_pool() {
 	rm -f "./newpool.yaml"
 }
 
+function create-block-ec-pool() {
```
Comment (nixpanic) [inline scripts/rook.sh]: function names may not contain `-`, use `_` here

File: scripts/rook.sh [hunk 4]
```diff
@@ -204,6 +221,12 @@ create-block-pool)
 delete-block-pool)
 	delete_block_pool
 	;;
+create-block-ec-pool)
```
Comment (nixpanic) [inline scripts/rook.sh:224]: case values (just strings) can contain `-`, so that is not a problem (as seen with other commands like `create-block-pool`).

**Review by nixpanic — CHANGES_REQUESTED** [review]

File: scripts/rook.sh (hunk 3 above)
Comment (Yuggupta27) [reply scripts/rook.sh]: Updated :+1:

File: scripts/rook.sh [hunk 5]
```diff
@@ -116,6 +117,22 @@ function delete_block_pool() {
 	rm -f "./newpool.yaml"
 }
 
+function create_block_ec_pool() {
+	curl -o block-pool-ec.yaml "${ROOK_URL}/pool-ec.yaml"
+	sed -i "s/ec-pool/$ROOK_BLOCK_EC_POOL_NAME/g" block-pool-ec.yaml
```
Comment (nixpanic) [inline scripts/rook.sh]: variables are used with `${..}` in this script, use `${ROOK_BLOCK_EC_POOL_NAME}` here and at other places

Comment (Yuggupta27) [reply scripts/rook.sh]: Thanks! Updated now

**Review by nixpanic — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Review by humblec — APPROVED** [review]

**Comment by Rakshith-R** [comment]
added DNM since #2658 needs to be merged first for the ci to work

**Pull request merged**